lib/core/unittest/CStringSimilarityTesterTest.cc

```diff
@@ -418,4 +348,56 @@ BOOST_AUTO_TEST_CASE(testWeightedEditDistance) {
+    ml::core::CStringSimilarityTester sst;
+
+    using TStrSizePr = std::pair<std::string, size_t>;
+    using TStrSizePrVec = std::vector<TStrSizePr>;
+
+    // These tests give a weight of 3 to dictionary words and 1 to other tokens
+    // along with positional weighting: x2 current weighting for first 4 tokens,
+    // x1 current weighting for next 3, x0 current weighting for the rest..
+    // We expect the resulting edit distance to differ from that where no positional
+    // weighting is in play, but still to be symmetric regarding the order of the strings passed.
+    TStrSizePrVec sourceShutDown1{
+        {"ml13", 2},   {"4608.1.p2ps", 2}, {"Info", 6},
+        {"Source", 6}, {"ML_SERVICE2", 1}, {"on", 3},
+        {"has", 3},    {"shut", 0},        {"down", 0}};
+    TStrSizePrVec sourceShutDown2{
+        {"ml13", 2},   {"4606.1.p2ps", 2}, {"Info", 6},
+        {"Source", 6}, {"MONEYBROKER", 1}, {"on", 3},
+        {"has", 3},    {"shut", 0},        {"down", 0}};
+
+    TStrSizePrVec serviceStart{
+        {"ml13", 2},    {"4606.1.p2ps", 2}, {"Info", 6},
+        {"Service", 6}, {"ML_FEED", 1},     {"id", 3},
+        {"of", 3},      {"has", 0},         {"started", 0}};
+    TStrSizePrVec noImageData{{"ml13", 2}, {"4607.1.p2ps", 2}, {"Debug", 6},
+                              {"Did", 6},  {"not", 1},         {"receive", 3},
+                              {"an", 3},   {"image", 0},       {"data", 0},
+                              {"for", 0},  {"ML_FEED", 0},     {"4205.T", 0},
+                              {"on", 0},   {"Recalling", 0},   {"item", 0}};
+    TStrSizePrVec empty;
+
+    BOOST_REQUIRE_EQUAL(size_t(3), sst.weightedEditDistance(sourceShutDown1, sourceShutDown2));
+    BOOST_REQUIRE_EQUAL(size_t(3), sst.weightedEditDistance(sourceShutDown2, sourceShutDown1));
+
+    BOOST_REQUIRE_EQUAL(size_t(3), sst.weightedEditDistance(sourceShutDown1, sourceShutDown2));
+    BOOST_REQUIRE_EQUAL(size_t(3), sst.weightedEditDistance(sourceShutDown2, sourceShutDown1));
+
+    BOOST_REQUIRE_EQUAL(size_t(15), sst.weightedEditDistance(sourceShutDown1, serviceStart));
+    BOOST_REQUIRE_EQUAL(size_t(15), sst.weightedEditDistance(serviceStart, sourceShutDown1));
+
+    BOOST_REQUIRE_EQUAL(size_t(13), sst.weightedEditDistance(sourceShutDown2, serviceStart));
+    BOOST_REQUIRE_EQUAL(size_t(13), sst.weightedEditDistance(serviceStart, sourceShutDown2));
+
+    BOOST_REQUIRE_EQUAL(size_t(21), sst.weightedEditDistance(noImageData, serviceStart));
+    BOOST_REQUIRE_EQUAL(size_t(21), sst.weightedEditDistance(serviceStart, noImageData));
+
+    BOOST_REQUIRE_EQUAL(size_t(21), sst.weightedEditDistance(noImageData, sourceShutDown1));
+    BOOST_REQUIRE_EQUAL(size_t(21), sst.weightedEditDistance(sourceShutDown1, noImageData));
+
+    BOOST_REQUIRE_EQUAL(size_t(23), sst.weightedEditDistance(serviceStart, empty));
+    BOOST_REQUIRE_EQUAL(size_t(23), sst.weightedEditDistance(empty, serviceStart));
```

Comment:
You don't need to cast the expected values with Boost.Test.  (The places where we do this are left over from the cppunit days.)